File: problema 12/c-languaje/problema_12.c
#include <stdio.h>

int main() {
    int vector[10];
    int i, temp;

    // Lectura de los elementos del vector
    printf("Ingrese 10 elementos para el vector:\n");
    for (i = 0; i < 10; i++) {
        scanf("%d", &vector[i]);
    }

    // Intercambio de elementos
    for (i = 0; i < 5; i++) {
        temp = vector[i];
        vector[i] = vector[9 - i];
        vector[9 - i] = temp;
    }

    // Impresión del vector modificado
    printf("Vector modificado:\n");
    for (i = 0; i < 10; i++) {
        printf("%d ", vector[i]);
    }
    printf("\n");

    return 0;
}

File: problema 12/fortran/problema_12.f90
program reverse_vector
    implicit none
    integer :: vector(10)
    integer :: i, temp

    ! Lectura de los elementos del vector
    print *, 'Ingrese 10 elementos para el vector:'
    do i = 1, 10
        read *, vector(i)
    end do

    ! Intercambio de elementos
    do i = 1, 5
        temp = vector(i)
        vector(i) = vector(11 - i)
        vector(11 - i) = temp
    end do

    ! Impresión del vector modificado
    print *, 'Vector modificado:'
    do i = 1, 10
        print *, vector(i)
    end do
end program reverse_vector